content/artifacts/artifact-automove/plugin.js

```diff
@@ -0,0 +1,387 @@
+        const candidate_rips = rips
+        .filter(p => (
+            p.planetLevel > artifact.rarity
+        ))
+        .map(to => {
+            return [to, distance(artifactPlanet, to)];
+        })
+        .sort((a, b) => a[1] - b[1]);
+
+        if(candidate_rips.length == 0) {
+          console.log(fromPlanet.locationId + ' can not automove to any spacetime rip');
+          continue;
+        }
+
+        // Remember its a tuple of candidates and their distance
+        const candidate_rip = candidate_rips[0][0];
+        // move artifact to the nearest spacetime rip.
+        n += this.move(artifactPlanet, artifactId, candidate_rip);
+      }
+
+      // move one artifact at a time!!!
+      break;
+    }
+    return n;
+  }
+
+
+  globalAutoMove() {
+    let n = 0;
+    let rips = df.getMyPlanets()
+    .filter(p => (
+        isSpaceRip(p)
+    ));
+
+    for (let planet of df.getMyPlanets()) {
+
+      if (planet.heldArtifactIds.length != 0) {
+        n += this.automove(planet, rips);
+        if (n >= this.conCurrentNum) {
+          break;
+        }
+      }
+    }
+  }
+
+
+  render(container) {
+    container.parentElement.style.minHeight = 'unset';
+    container.style.minHeight = 'unset';
+
+    container.style.width = '260px';
+
+    let stepperLabel = document.createElement('label');
+    stepperLabel.innerText = 'Max % energy to spend';
+    stepperLabel.style.display = 'block';
+
+    let stepper = document.createElement('input');
+    stepper.type = 'range';
+    stepper.min = '0';
+    stepper.max = '100';
+    stepper.step = '5';
+    stepper.value = `${this.maxEnergyPercent}`;
+    stepper.style.width = '80%';
+    stepper.style.height = '24px';
+
+    let percent = document.createElement('span');
+    percent.innerText = `${stepper.value}%`;
+    percent.style.float = 'right';
+
+    stepper.onchange = (evt) => {
+      percent.innerText = `${evt.target.value}%`;
+      try {
+        this.maxEnergyPercent = parseInt(evt.target.value, 10);
+      } catch (e) {
+        console.error('could not parse energy percent', e);
+      }
+    }
+
+    let levelLabel = document.createElement('label');
+    levelLabel.innerText = 'Max. level to use';
+    levelLabel.style.display = 'block';
+
+    let level = document.createElement('select');
+    level.style.background = 'rgb(8,8,8)';
+    level.style.width = '100%';
+    level.style.marginTop = '10px';
+    level.style.marginBottom = '10px';
+    planetLevels.forEach(lvl => {
+      let opt = document.createElement('option');
+      opt.value = `${lvl}`;
+      opt.innerText = `Level ${lvl}`;
+      level.appendChild(opt);
+    });
+    level.value = `${this.maxMiddlePlanetLevel}`;
+
+    level.onchange = (evt) => {
+      try {
+        this.maxMiddlePlanetLevel = parseInt(evt.target.value, 10);
+      } catch (e) {
+        console.error('could not parse planet level', e);
+      }
+    }
+
+    let deactiveArtifactLabel = document.createElement('label');
```

Comment:
Fix typo - please can we update `deactiveArtifactLabel` to `deactivateArtifactLabel`? 5 instances in file.